File: Oauth2Simples/src/main/java/br/com/estudos/oauth2/service/StocksService.java
package br.com.estudos.oauth2.service;

import br.com.estudos.oauth2.model.Stocks;

import org.springframework.stereotype.Service;
import org.springframework.beans.factory.annotation.Autowired;
import br.com.estudos.oauth2.repository.StocksRepository;

import java.util.List;
import java.util.Optional;

@Service
public class StocksService{


    @Autowired
    private StocksRepository repository;

    
    public List<Stocks> findAll(){
        return repository.findAll();
    }

    
    public Optional<Stocks> findByTicket(String ticket){
        return repository.findByTicket(ticket);
    }

}

File: kotlin/kotlinEstudos/src/main/java/br/com/kotlin/estudos/services/StocksService.kt
package br.com.kotlin.estudos.services

import br.com.kotlin.estudos.model.repository.financial.Stock
import br.com.kotlin.estudos.repository.StocksRepository
import org.springframework.stereotype.Service

@Service
class StocksService(
        private val stocksRepository: StocksRepository
) {
    fun findAll(): List<Stock> = stocksRepository.findAll().toList()
}